### PR TITLE
RA-1795: <encounterDiagnoses/> to skip voided diagnoses.

### PR DESCRIPTION
**Pull request by icrc-psousa**
Related ticket: https://issues.openmrs.org/browse/RA-1795

### CODE REVIEW

File: api-2.2/src/main/java/org/openmrs/module/coreapps/htmlformentry/EncounterDiagnosesElement.java [hunk 1]
```diff
@@ -378,10 +378,16 @@ public void applyAction(FormEntrySession formEntrySession) {
...
-        	}
+            Encounter existingEncounter = context.getExistingEncounter();
+            if (existingEncounter != null) {
+                Set<org.openmrs.Diagnosis> nonVoidedDiagnoses = new HashSet<org.openmrs.Diagnosis>();
+                for (org.openmrs.Diagnosis diagnosis : existingEncounter.getDiagnoses()) {
+                    if (!diagnosis.getVoided()) {
+                        nonVoidedDiagnoses.add(diagnosis);
+                    }
+                }
+                return nonVoidedDiagnoses;
```
Comment (mks-d) [inline api-2.2/src/main/java/org/openmrs/module/coreapps/htmlformentry/EncounterDiagnosesElement.java:389]: I was hoping we could do this:
```java
return existingEncounter.getDiagnoses().stream().filter(d -> !d.getVoided()).collect(Collectors.toSet());
```
But for this to work we would need to put the JDK compliance to 1.8 (_just_ for the api-2.2 submodule). Do you think you could try that out?

Comment (icrc-psousa) [reply api-2.2/src/main/java/org/openmrs/module/coreapps/htmlformentry/EncounterDiagnosesElement.java:389]: I tried just to specify JDK 1.8 in the api-2.2 module but, all existing tests relying Spring ApplicationContext began to fail. It seems to be using Spring 3 thus failing to load Java 8 classes.

```
java.lang.IllegalStateException: Failed to load ApplicationContext
(...)
Caused by: org.springframework.core.NestedIOException: ASM ClassReader failed to parse class file - probably due to a new Java class file version that isn't supported yet
```

Comment (icrc-psousa) [reply api-2.2/src/main/java/org/openmrs/module/coreapps/htmlformentry/EncounterDiagnosesElement.java]: @mks-d, api-2.2 submodule is now using JDK 8 and I've reviewed the code to use streams.

**Comment by icrc-psousa** [comment]
@mks-d, actually I had updated the pull request before testing openmrs at runtime.
When starting openmrs with the updated module, ApplicationContext seems to be failing to initializing. I'll still take a more careful look.

`
openmrs-web_1    | Related cause: org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'appFrameworkService' defined in URL [jar:file:/opt/openmrs/.openmrs-lib-cache/appframework/lib/appframework-api-2.16.0-SNAPSHOT.jar!/moduleApplicationContext.xml]: Cannot create inner bean 'org.openmrs.module.appframework.service.AppFrameworkServiceImpl#350f5214' of type [org.openmrs.module.appframework.service.AppFrameworkServiceImpl] while setting bean property 'target'; nested exception is org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'org.openmrs.module.appframework.service.AppFrameworkServiceImpl#350f5214' defined in URL [jar:file:/opt/openmrs/.openmrs-lib-cache/appframework/lib/appframework-api-2.16.0-SNAPSHOT.jar!/moduleApplicationContext.xml]: Cannot resolve reference to bean 'allComponentsState' while setting constructor argument; nested exception is org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'allComponentsState' defined in URL [jar:file:/opt/openmrs/.openmrs-lib-cache/appframework/lib/appframework-api-2.16.0-SNAPSHOT.jar!/moduleApplicationContext.xml]: Initialization of bean failed; nested exception is org.springframework.aop.framework.AopConfigException: Unexpected AOP exception; nested exception is java.lang.ClassCastException: org.openmrs.module.appframework.repository.AllComponentsState$$EnhancerBySpringCGLIB$$9916589d cannot be cast to org.springframework.cglib.proxy.Factory
`

**Pull request merged**